stl/inc/numeric

```diff
@@ -490,17 +490,19 @@ _CONSTEXPR20 _OutIt adjacent_difference(const _InIt _First, const _InIt _Last, _
 #endif
             {
                 // Go with pointers and without loop-carried dependency to enable vectorization
-                const auto _Count         = _ULast - _UFirst;
-                const auto _Count_as_size = static_cast<size_t>(_Count);
-                const auto _Src_ptr       = _STD _To_address(_UFirst);
-                const auto _Dest_ptr      = _STD _To_address(_UDest);
+                const auto _First_ptr = _STD _To_address(_UFirst);
+                const auto _Last_ptr  = _STD _To_address(_ULast);
+                const auto _Dest_ptr  = _STD _To_address(_UDest);
+                const auto _Count     = _Last_ptr - _First_ptr;
+
                 // Need to perform aliasing analysis.

```

Comment:
I don't like having to wotk with count as signed, as it might overflow by pointer range that is more than half of 32 bit address space. That's fine for the loop where overlaps are checked, as then we certainly don't have such big range, but not sure about the check itself

Reply:
This was a pretty hasty hack; if you'd like to perform a followup to use `size_t` I'd be fine with that.

Reply:
I don't know honestly, after giving this more thought.

The subtraction of such pointers already an UB. I thought we should immediate cast to `size_t` and further operate on that type, but... is it... less UB?

And if we are fine with UB that happens to work then it is likely that comparison of pointers with added overflown size will overflow again, ultimately resulting in the right value.

Knowing the pointer representation we can enter the implementation defined territory and just cast them to `uintptr_t` and get size this way, but... are we willing to do this everywhere?

I think this is somewhat similar to situation of `shared_ptr` counter overflow. It seems that an obscure situation is legal and should be handled, but it is impractical to do that.